app/models/schedule.rb

```diff
@@ -0,0 +1,26 @@
+class Schedule < ActiveRecord::Base
+  has_many :bookable_slots, -> { order(:date) }
+
+  alias default? new_record?
+
+  def generate_bookable_slots!
+    BookableSlotGenerator.new(self).call
+  end
+
+  def bookable_slots_in_window
+    bookable_slots.windowed
+  end
+
+  def self.current(location_id)
+    where(location_id: location_id)
+      .order(created_at: :desc)
+      .first_or_initialize(location_id: location_id)
+  end
+
+  def self.slot_attributes
```

Comment:
Not sure about this, maybe just a heap easier to explicitly define these attributes as a constant array of symbols instead. 

The purpose of this is to easily and collectively refer to the availability associated attributes defined by the schedule - for example `monday_am`, `monday_pm`, `tuesday_am`... etc.

We needed to jump through some hoops in the factory to accommodate too, given those attributes are not defined during migration.